pytealext/auto_load_scratch_var.py

```diff
@@ -0,0 +1,70 @@
+from pyteal import Bytes, CompileOptions, Expr, Int, ScratchVar, TealBlock, TealSimpleBlock, TealType
+
+
+class AutoLoadScratchVar(Expr):
+    """Makes ScratchVars more convenient to use.
+
+    When used inline with another expression, a load is performed automatically
+    without the need for explicit load() calls.
+
+    Store still has to be called manually.
+
+    Example:
+    ```
+        s = AutoLoadScratchVar(TealType.uint64)
+        sum = AutoLoadScratchVar(TealType.uint64)
+        program = For(s.store(Int(0)), s < Int(10), s.increment()).Do(
+            sum.increment(s)
+        )
+    ```
+    """
+
+    def __init__(self, _type: TealType = TealType.anytype, slotId: int = None):  # pylint: disable=redefined-builtin
```

Comment:
why `# pylint: disable=redefined-builtin` is needed here? `_type` instead of `type` doesn't work?

Reply:
it should be type to match pyteal

Reply:
unfortunately